pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -244,7 +244,7 @@ public boolean isMutableSegment() {
     _logger =
         LoggerFactory.getLogger(MutableSegmentImpl.class.getName() + "_" + _segmentName + "_" + config.getStreamName());
 
-    Set<String> noDictionaryColumns = config.getNoDictionaryColumns();
+    Set<String> noDictionaryColumns = new HashSet<>(config.getNoDictionaryColumns());
```

Comment:
nit: I know it's because the elements of this set can be removed down below but it'd be good to add some comments here to explain why the copy of `noDictionaryColumns` is made here.

Reply:
done